spec/Section 3 -- Type System.md

```diff
@@ -1778,3 +1790,36 @@ type ExampleType {
   oldField: String @deprecated(reason: "Use `newField`.")
 }
 ```
+
+
+### @specifiedBy
+
+```graphql
+directive @specifiedBy(
+  url: String!
+) on SCALAR
+```
+
+The `@specifiedBy` directive is used within the type system definition language
+to provide a URL for specifying the behaviour of custom
+scalar definitions. The URL should point to a human-readable specification of
+the data format, serialization, and coercion rules for the scalar. For example,
+a GraphQL system providing a `UUID` scalar might link to RFC 4122,
+or some document defining a reasonable subset of that RFC. If a specification
+URL is present, systems must conform to the described rules. Built-in scalar
+types should not provide a URL in this way.
+
+Specifications linked in this way should provide a single, stable scalar format
+to avoid ambiguity. This means that if the specification is in flux, the system
+should link to a fixed version rather than to a resource whose contents might
+change. It is also recommended that the URL associated with a scalar not be
+changed once defined; doing so is likely to disrupt tooling, and may come with
+changes to the specification contents which are themselves breaking.
+
+In this example, two custom scalar types are defined with URLs pointing to the
+relevant IETF specifications.
+
+```graphql example
+scalar UUID @specifiedBy(url: "https://tools.ietf.org/html/rfc4122")
```

Comment:
While implementing this in https://github.com/graphql/graphql-js/pull/2276 @IvanGoncharov  provided the following [comment](https://github.com/graphql/graphql-js/pull/2276#discussion_r358119871):
> If at some point we decide to add `UUID` scalar into this library it would be a confusing example.
> It's better to use some abstract scalar like `Foo` and `https://example.com/foo_spec`.

**Question:** Do we want to make that same change here as well?  If so this likely applies to `URL` as well, and effects a few locations in this PR.

Reply:
Point taken that it could become confusing, but it's not confusing yet, and there is definitely value in having a concrete example that lines up with use cases a lot of actual developers will run into. I'm planning to leave this as is, and we can change it if becomes an issue later.